cmake/test_setup_prologue.cmake

```diff
@@ -0,0 +1,53 @@
+# This is a target file designed to run on a host system.
+
+# Default coverage option to off.
+option(COVERAGE "COVERAGE" OFF)
+
+# Enable the required language standards.
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS OFF)
+
+if(NOT DEFINED GTEST_TAG)
+set(GTEST_TAG v1.15.2)
+endif()
+message(STATUS "Using GoogleTest tag: ${GTEST_TAG}")
+
+# Get a copy of GoogleTest.
+include(FetchContent)
+FetchContent_Declare(
+    googletest
+    URL https://github.com/google/googletest/archive/refs/tags/${GTEST_TAG}.zip
+)
+
+# On Windows: Prevent overriding the parent project's compiler/linker settings.
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+# This magic is required in order to properly add the "-x" option for the
```

Comment:
why don't we just add a compile option -x c++ to the c++ files?

Reply:
I will keep this as is, except there appears to be an extra "-c" option, which I removed.